src/core/server/saved_objects/saved_objects_service.ts

```diff
@@ -283,6 +283,14 @@ export class SavedObjectsService
     const cliArgs = this.coreContext.env.cliArgs;
     const skipMigrations = cliArgs.optimize || savedObjectsConfig.skip;
 
+    this.logger.debug(
+      'Waiting until all Elasticsearch nodes are compatible with Kibana before starting saved objects migrations...'
+    );
+    await this.setupDeps!.elasticsearch.esNodesCompatibility$.pipe(
```

Comment:
This behaviour isn't 100% the same as what it was in legacy. In legacy we would start the status service so even though migrations wouldn't run, there would be a running server which showed that the Elasticsearch plugin was red with the reason which helps surface the underlying problem. Once we have a status service in NP we should aim to create similar behaviour.